### DISCUSSION TITLE
Multiscale microstructure

### DISCUSSION THREAD

**Discussion by J-AKAZ** [neperfepx/neper#238] [Q&A] [answered]
Hello Romain,
I want to create a microstructure using the following EBSD map bellow : 
![Fe-BCC](https://user-images.githubusercontent.com/79088662/120290684-dfda4600-c2c2-11eb-8e90-72844ace6327.png)
I divided this map in four parts. And and I created with Neper a 4 part vornoi microstructure using this command: 
`neper -T -n 4 -domain 'cube(0.03,0.08,0.03)' -morphooptiini "coo:file(coo1.txt),weight:0" -morpho "voronoi" -oridescriptor 'euler-roe' -format tess -reg 1 -o r1  `
the _coo1_ file is given bellow : 
[coo1.txt](https://github.com/rquey/neper/files/6574645/coo1.txt)
And I obtained the following microstructure : 
![r1](https://user-images.githubusercontent.com/79088662/120291622-cbe31400-c2c3-11eb-8484-e9f61e2186b1.png)
Then in each part I create the number of daughter grain that corresponds to it : 
`neper -T -n "4::file(rafinnement.txt)" -domain 'cube(0.03,0.08,0.03)' -morphooptiini "coo:file(coo1.txt),weight:0" -morpho "voronoi" -oridescriptor 'euler-roe' -format tess,ori -reg 1 -o r2`
The _rafinnement.txt_ is given bellow : 
[rafinnement.txt](https://github.com/rquey/neper/files/6574686/rafinnement.txt)
And I obtained the following microstructure : 
![r2](https://user-images.githubusercontent.com/79088662/120292867-f6819c80-c2c4-11eb-8177-33a9436eb63c.png)
the next step is to create blades inside each grain. I used the following command : 
`neper -T -n "4::from_morpho::file(raffinement.txt)" -domain 'cube(0.03,0.08,0.03)' -morphooptiini "coo:file(coo1.txt),weight:0" -morpho "voronoi::sphericity:lognormal(0.1,0.01)::lamellar(w=0.001)" -oridescriptor 'euler-roe' -format tess,vtk,ori -o D522_3_1`
But it doesn't work. I get the terminal Neper output : 
`========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
Info   : Running on 12 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/g20281/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 4::from_morpho::file(raffinement.txt) -domain
         cube(0.03,0.08,0.03) -morphooptiini coo:file(coo1.txt),weight:0
         -morpho
         voronoi::sphericity:lognormal(0.1,0.01)::lamellar(w=0.001)
         -oridescriptor euler-roe -format tess,vtk,ori -o D522_3_1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Error  :   - `-morpho lamellar(w=0.001)' requires `-n from_morpho'.
Abandon
`
Could you help me please or give me some suggestions to create a similar microstructure as the EBSD map?
Thank you so much,

**Comment by rquey** [comment, marked as answer]
Sorry for the delayed response, but I would start by merging the 4 EBSD maps, as this is not a multiscale microstructure _per se_...
